src/cargo/ops/cargo_update.rs

```diff
@@ -795,6 +743,20 @@ fn status_locking(ws: &Workspace<'_>, num_pkgs: usize) -> CargoResult<()> {
     Ok(())
 }
 
+fn report_latest(possibilities: &[IndexSummary], package: PackageId) -> Option<String> {
+    if possibilities.is_empty() {
+        return None;
+    }
```

Comment:
nit: [`Iterator::max`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.max) returns `None` when empty. So this is kinda redundant?